optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -159,10 +160,27 @@ private void processEntityAnnotations(DescriptorPolicy descriptorPolicy) {
 
     private void processMovable(DescriptorPolicy descriptorPolicy, PlanningEntity entityAnnotation) {
         Class<? extends SelectionFilter> movableEntitySelectionFilterClass = entityAnnotation.movableEntitySelectionFilter();
-        if (movableEntitySelectionFilterClass == PlanningEntity.NullMovableEntitySelectionFilter.class) {
-            movableEntitySelectionFilterClass = null;
-        }
-        if (movableEntitySelectionFilterClass != null) {
+        boolean hasSelectionFilter = movableEntitySelectionFilterClass != PlanningEntity.NullMovableEntitySelectionFilter.class;
+        Class<? extends PinningFilter> pinningFilterClass = entityAnnotation.pinningFilter();
+        boolean hasPinningFilter = pinningFilterClass != PlanningEntity.NullPinningFilter.class;
+        if (hasPinningFilter && hasSelectionFilter) {
+            throw new IllegalStateException("The entityClass (" + entityClass
+                    + ") uses both movableEntitySelectionFilter (" + movableEntitySelectionFilterClass +
+                    ") and pinningFilter (" + pinningFilterClass + "). " +
+                    "Use pinningFilter exclusively.");
```

Comment:
Our error messages typically have 2 lines:
- line 1: what is wrong. This is always truth. It doesn't say how to fix it though. Ends with "\n"
- line 2 (optional): How to fix it. This isn't always the way to fix it, it is a recommendation. Therefore, it starts with "Maybe ..."

I'd also add a breadcrumb to @PlanningEntity

=>
```
            throw new IllegalStateException("The entityClass (" + entityClass
                    + ") uses both movableEntitySelectionFilter (" + movableEntitySelectionFilterClass +
                    ") and pinningFilter (" + pinningFilterClass + ").\n" +
                    "Maybe use only pinningFilterClass on your @" + PlanningEntity.class.getSimpleName() + " annotation.");
```